tests/actions.go

```diff
@@ -349,10 +353,44 @@ func (oa *operatorActions) StopInsertDataTo(info *TidbClusterInfo) error {
 	return nil
 }
 
-func (oa *operatorActions) ScaleTidbCluster(info *TidbClusterInfo) error   { return nil }
-func (oa *operatorActions) UpgradeTidbCluster(info *TidbClusterInfo) error { return nil }
-func (oa *operatorActions) DeployMonitor(info *TidbClusterInfo) error      { return nil }
-func (oa *operatorActions) CleanMonitor(info *TidbClusterInfo) error       { return nil }
+func chartPath(name string, tag string) string {
+	return "/charts/" + tag + "/" + name
+}
+
+func (oa *operatorActions) ScaleTidbCluster(info *TidbClusterInfo) error {
+	cmd := fmt.Sprintf("helm upgrade %s %s --set-string %s",
+		info.ClusterName, chartPath("tidb-cluster", info.OperatorTag), info.HelmSetString())
+	glog.Info("[SCALE] " + cmd)
+	res, err := exec.Command("/bin/sh", "-c", cmd).CombinedOutput()
+	if err != nil {
+		return errors.Wrapf(err, "failed to scale tidb cluster: %s", string(res))
+	}
+	return nil
+}
+
+func (oa *operatorActions) UpgradeTidbCluster(info *TidbClusterInfo) error {
```

Comment:
change to `UpgradeTidbClusterTo(info *TidbClusterInfo,version string)`

Reply:
But how to upgrade only one component (eg. just upgrade tidb) ?